ClientManagementService/ClientManagementService.API/Controllers/ClientController.cs

```diff
@@ -1,15 +1,156 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
+using ClientManagementService.API.DTO;
+using ClientManagementService.API.DTOMapper;
+using ClientManagementService.Domain.Services;
 using Microsoft.AspNetCore.Mvc;
 
 namespace ClientManagementService.API.Controllers
 {
-    [Route("api/[controller]")]
+    [Route("api/client")]
     [ApiController]
     public class ClientController : ControllerBase
     {
+        private readonly IClientService _clientService;
+
+        public ClientController(IClientService clientService)
+        {
+            _clientService = clientService;
+        }
+
+        [HttpPost("create")]
+        public async Task<IActionResult> CreateClient([FromBody] ClientDTO client)
+        {
+            try
+            {
+                await _clientService.CreateClient(ClientDTOMapper.FromDTOClient(client), client.Password);
+
+                return StatusCode(201);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetClientById(long id)
+        {
+            try
+            {
+                var client = await _clientService.GetClientById(id);
+
+                return Ok(ClientDTOMapper.ToDTOClient(client));
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpGet("email/{email}")]
+        public async Task<IActionResult> GetClientByEmail(string email)
+        {
+            try
+            {
+                var client = await _clientService.GetClientByEmail(email);
+
+                return Ok(ClientDTOMapper.ToDTOClient(client));
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPatch("login")]
+        public async Task<IActionResult> ClientLogin([FromBody] ClientDTO client)
+        {
+            try
+            {
+                await _clientService.ClientLogin(client.EmailAddress, client.Password);

```

Comment:
We need to return some client info. Look at the employee management and see what we return.

I think minimum you need to return the Id and FirstName.